src/platform/unix/mod.rs

```diff
@@ -470,10 +470,21 @@ impl OsIpcReceiverSet {
 
     pub fn select(&mut self) -> Result<Vec<OsIpcSelectionResult>,UnixError> {
         let mut selection_results = Vec::new();
-        match self.poll.poll(&mut self.events, None) {
-            Ok(sz) if sz > 0 => {},
-            _ => { return Err(UnixError::last()); }
-        };
+        let mut sz = 0;
+        while sz <=0 {
+            match self.poll.poll(&mut self.events, None) {
+                Ok(num_events) => {
+                    if num_events > 0 {
+                        sz = num_events;
+                    }
```

Comment:
There is no need for this `if`: you are checking the same thing in the `while` condition.

(Or you could use a `loop` instead of `while`, and do an explicit `break` when getting results; thus keeping the condition here. Not sure though whether this would be clearer, or less clear...)